### ISSUE TITLE
[DraculaX] NBG3 needs to be scrolled -8 dot

### ISSUE THREAD

**Issue by devmiyax** [devmiyax/yabause#172]
![image](https://cloud.githubusercontent.com/assets/1174152/12761992/9bd083c4-ca30-11e5-89a4-719fbbf8d21a.png)

**Comment by devmiyax** [comment]
Vdp2Regs->SCXN3 = 0x200
 Vdp2Regs->SCYN3 = 0x008


4-bit(16 colors)
Tile(1H x 1V)
Plane Size = 1H x 1V
Pattern Name data size = 1 word
Character Number Supplement bit = 0
Special Priority bit = 0
Special Color Calculation bit = 0
Supplementary Palette number = 2
Supplementary Color number = 2
Plane A Address = 00050000
Plane B Address = 00052000
Plane C Address = 00054000
Plane D Address = 00056000
Screen Scroll x = 0, y = -8
Window W0 Enabled:
Horizontal start = 0
Vertical start = 12
Horizontal end = 320
Vertical end = 240
Display inside of Window
Window Overlap Logic: AND
Color Ram Address Offset = 0
Priority = 4
Color Calculation Enabled
Extended Color Calculation Enabled
Color Calculation Ratio = 31:1
Color Offset A Enabled
R = 0, G = 0, B = 0
Special Color Calculation 0

**Comment by devmiyax** [comment]
write register value at PC 0x0624b5a


-		sh->regs.R	0x0ba00560 {0x06024a7a, 0x25f80094, 0x06065644, 0x00000200, 0x25f80070, 0x06065620, 0x00000040, 0x25f80000, ...}	unsigned int[0x00000010]
		[0x00000000]	0x06024a7a	unsigned int
		[0x00000001]	0x25f80094	unsigned int
		[0x00000002]	0x06065644	unsigned int
		[0x00000003]	0x00000200	unsigned int
		[0x00000004]	0x25f80070	unsigned int
		[0x00000005]	0x06065620	unsigned int
		[0x00000006]	0x00000040	unsigned int
		[0x00000007]	0x25f80000	unsigned int
		[0x00000008]	0x06024b20	unsigned int
		[0x00000009]	0x0000000f	unsigned int
		[0x0000000a]	0x00000000	unsigned int
		[0x0000000b]	0x00000000	unsigned int
		[0x0000000c]	0x00000000	unsigned int
		[0x0000000d]	0x00000000	unsigned int
		[0x0000000e]	0x06001f2c	unsigned int
		[0x0000000f]	0x06001f2c	unsigned int

**Comment by BenjaminSiskoo** [comment]
There is a fix to the software version. https://github.com/Yabause/yabause/pull/153

**Comment by devmiyax** [comment]
I don't think Yabause#153 is good.

**Comment by mishamosher** [comment]
This doesn't happen in [Mednafen](https://mednafen.github.io/). Maybe something from their source code could help?

**Comment by devmiyax** [comment]
I know why it happens and how to fix it. I just care about performance..

**Comment by devmiyax** [comment]
![image](https://user-images.githubusercontent.com/1174152/82811024-c857c880-9eca-11ea-840f-8de8d9f10f1b.png)
![image](https://user-images.githubusercontent.com/1174152/82811036-ce4da980-9eca-11ea-926a-dbcbef0010f2.png)
![image](https://user-images.githubusercontent.com/1174152/82811045-d3aaf400-9eca-11ea-8dcf-53faea24aeaa.png)

**Comment by devmiyax** [comment]
![image](https://user-images.githubusercontent.com/1174152/82811090-ea514b00-9eca-11ea-8b5f-784faa7409d4.png)

**Comment by devmiyax** [comment]
the real machine CYCA0L=0xfe67
![2020052520550849](https://user-images.githubusercontent.com/1174152/82811138-05bc5600-9ecb-11ea-811b-df95a9930984.png)

**Comment by devmiyax** [comment]
the real machine CYCA0L=0x67fe
![2020052521033574](https://user-images.githubusercontent.com/1174152/82811245-3f8d5c80-9ecb-11ea-8873-f3b6f94db4a9.png)

**Comment by devmiyax** [comment]
The changes in software version is not good.
https://github.com/Yabause/yabause/pull/153/commits/1b68e77d7ac81a5c7ca0fffad3989337fc6cb23b#diff-12bc0117b153607d0d5eaacefb350bf0R3435

**Comment by devmiyax** [comment]
This issue is caused by VDP2 VRAM cycle pattern miss-setting.
The setting of DraculaX is restricted.
In Fighters megamix, char data access is earlier than pattern name access, so next char access is postponed.

**Comment by devmiyax** [comment]
Fighters Megamix issue is happen only when hi-res mode
![2020052523275327](https://user-images.githubusercontent.com/1174152/82821667-740b1380-9edf-11ea-8692-d995b6946970.png)

**Comment by fafling** [comment]
In Yaba Sanshiro v3.2.4, the offset of NBG3 is indeed fixed in the screens following the title in Akumajou Dracula X, but it is still there in the in-game dialogs which are displayed by NBG0.
![image](https://user-images.githubusercontent.com/7624262/83335498-83a6a580-a2ad-11ea-8d92-53374c45d86f.png)

Display on the console :
![image](https://user-images.githubusercontent.com/7624262/83335558-062f6500-a2ae-11ea-90bb-d514b786d602.png)

I checked the cycle pattern register value with the memory editor (address 05F80010). The value during the screens following the title is :
6666FFFF FFFFFFFF FFFFFFFF FFFFFFFF
The value during the 1st dialog is :
7F7FFFFF FFFFFFFF 3F3FFFFF FFFFFFFF
It doesn't seem that these values are accurate with so little timings set to access.

Save state :
[Akumajou_dracula_X_1st_dialog.zip](https://github.com/devmiyax/yabause/files/4706163/Akumajou_dracula_X_1st_dialog.zip)

Tested with the original CD.

I also wanted to check Fighters megamix, but it makes a black screen after Sega logo (tested with US CD).
